### PR TITLE
Fix some flakes in autoscaler e2e on gke

### PR DESCRIPTION
**Pull request by MaciekPytel**
This PR should fix some of the flakes we found in e2e runs, while testing for 1.7 release:
 - if one of the nodes is unschedulable in set up (causing set up to fail) we used to wait for wrong number of nodes in clean-up, adding unnecessary 20 minute wait to failing test
 - we did not check for errors when creating RC in test, leading to tests failing later in hard to debug way (added retry loop and explicit test failure)

### CODE REVIEW

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -48,13 +48,15 @@ import (
...
-	nodesRecoverTimeout   = 5 * time.Minute
+	defaultTimeout         = 3 * time.Minute
+	resizeTimeout          = 5 * time.Minute
+	scaleUpTimeout         = 5 * time.Minute
+	scaleUpTriggerTimeout  = 2 * time.Minute
+	scaleDownTimeout       = 20 * time.Minute
+	podTimeout             = 2 * time.Minute
+	nodesRecoverTimeout    = 5 * time.Minute
+	rcCreationRetryTimeout = 4 * time.Minute
+	rcCreationRetryDelay   = 1 * time.Minute
```
Comment (mwielgus) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: 20 sec?

**Comment by mwielgus** [comment]
One nit, otherwise LGTM

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 1 above)
Comment (MaciekPytel) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: done

**Pull request merged**